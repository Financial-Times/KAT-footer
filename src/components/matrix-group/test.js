import React from 'react';
import { expect } from 'chai';
import { mount } from 'enzyme';
import MatrixGroup from './index.js';

describe('MatrixGroup', () => {

  describe('component', () => {
    const i = 0;
    const link1 = {
      href: 'web.link',
      test: 'test'
    };
    const link2 = {
      href: 'web.link',
      test: 'test'
    };
    const data = {
      title: 'Test',
      items: [ link1, link2 ]
    };
    const wrapper = mount(<MatrixGroup index={i} groupData={data}/>);

    test('has matrix title', () => {
      const content = <h6 className="o-footer__matrix-title">
        {data.title}
      </h6>;
      expect(wrapper.contains(content)).to.equal(true);
    });

    test('has footer matrix group', () => {
      expect(wrapper.find('.o-footer__matrix-group').length).to.equal(1);
    });

    test('has footer matrix content', () => {
      expect(wrapper.find('.o-footer__matrix-content').length).to.equal(1);
    });

    test('has footer matrix column', () => {
      expect(wrapper.find('.o-footer__matrix-column').length).to.equal(1);
    });

    test('has matrix link', () => {
      const content = <a className="o-footer__matrix-link" href={link1.href} key={0}>
        {link1.text}
      </a>;
      expect(wrapper.contains(content)).to.equal(true);
    });

    test('has right number of matrix links', () => {
      expect(wrapper.find('.o-footer__matrix-link').length).to.equal(data.items.length);
    });
  });
});
