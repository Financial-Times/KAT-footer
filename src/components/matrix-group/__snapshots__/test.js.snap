// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatrixGroup component matches snapshot 1`] = `
'<div className=\\'o-footer__matrix-group o-footer__matrix-group--1\\'>
<h6 className=\\'o-footer__matrix-title\\'>
Test
</h6>
<div className=\\'o-footer__matrix-content\\' id=\\'o-footer-section-0\\'>
<div className=\\'o-footer__matrix-column\\'>
<a className=\\'o-footer__matrix-link\\' href=\\'web.link\\' />
<a className=\\'o-footer__matrix-link\\' href=\\'web.link\\' />
</div>
</div>
</div>'
`;
